lte/gateway/python/magma/mobilityd/tests/test_dhcp_helper_cli.py

```diff
@@ -0,0 +1,131 @@
+"""
+Copyright (C) 2022  The Magma Authors
+
+This program is free software; you can redistribute it and/or
+modify it under the terms of the GNU General Public License
+as published by the Free Software Foundation; either version 2
+of the License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; If not, see <http://www.gnu.org/licenses/>.
+"""
+from typing import Optional
+from unittest.mock import patch
+
+import pytest
+from scapy.layers.dhcp import BOOTP, DHCP
+from scapy.layers.l2 import Dot1Q, Ether
+from scripts.dhcp_helper_cli import (
+    DhcpHelperCli,
+    DHCPState,
+    MacAddress,
+    release_arg_handler,
+)
+
+MACSTRING = "12:34:56:78:90:AB"
+MAC = MacAddress(MACSTRING)
+VLAN = 0
+IFACE = "dhcp0"
+IPv4 = 0x0800
+
+DHCP_OFFER_PKT = Ether(src="00:00:00:00:00:00", dst=MAC)
+DHCP_OFFER_PKT /= BOOTP(yiaddr="1.2.3.4", chaddr=MAC.as_hex())
+DHCP_OFFER_PKT /= DHCP(options=[("message-type", DHCPState.OFFER), "end"])
+
+DHCP_ACK_PKT = Ether(src="00:00:00:00:00:00", dst=MAC)
+DHCP_ACK_PKT /= BOOTP(yiaddr="1.2.3.4", chaddr=MAC.as_hex())
+DHCP_ACK_PKT /= DHCP(options=[("message-type", DHCPState.ACK), "end"])
+
+DHCP_RELEASE_PKT = Ether(src="00:00:00:00:00:00", dst=MAC)
+DHCP_RELEASE_PKT /= BOOTP(yiaddr="1.2.3.4", chaddr=MAC.as_hex(), ciaddr="4.5.6.7")
+DHCP_RELEASE_PKT /= DHCP(options=[("message-type", DHCPState.RELEASE), "end"])
+
+
+@pytest.fixture()
+def dhcp_helper_cli_fixture(mac: MacAddress = MAC, vlan: int = VLAN, iface=IFACE, ip: Optional[str] = None, server_ip: Optional[str] = None):
+    with patch("scripts.dhcp_helper_cli.AsyncSniffer"):
+        return DhcpHelperCli(mac, vlan, iface, ip, server_ip)
+
+
+@patch("scripts.dhcp_helper_cli.sendp")
+def test_send_dhcp_discover(sendp_mock, dhcp_helper_cli_fixture):
+    dhcp_helper_cli_fixture.send_dhcp_discover()
+    pkt = sendp_mock.call_args[0][0]
+
+    assert pkt[Ether].type == IPv4
+    assert pkt[Ether].src == MACSTRING.lower()
+    assert Dot1Q not in pkt
+    assert ('message-type', 'discover') in pkt[DHCP].options
+
```

Comment:
Do we want to assert the DHCP state here as well?
```suggestion
assert dhcp_helper_cli_fixture._state == DHCPState.DISCOVER

```

Reply:
done